src/main/java/ru/sbt/mipt/oop/SmartHome.java

```diff
@@ -1,24 +1,55 @@
 package ru.sbt.mipt.oop;
 
+import ru.sbt.mipt.oop.roomIterator.RoomIterator;
+import ru.sbt.mipt.oop.roomIterator.RoomIteratorCollection;
+import ru.sbt.mipt.oop.roomIterator.RoomIteratorCollectionImpl;
+
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
 
-public class SmartHome {
-    Collection<Room> rooms;
+public class SmartHome  implements Action {
+    List<Room> rooms;
 
     public SmartHome() {
         rooms = new ArrayList<>();
     }
 
-    public SmartHome(Collection<Room> rooms) {
+    public SmartHome(List<Room> rooms) {
         this.rooms = rooms;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        SmartHome smartHome = (SmartHome) o;
+        return Objects.equals(rooms, smartHome.rooms);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(rooms);
+    }
+
     public void addRoom(Room room) {
         rooms.add(room);
     }
 
     public Collection<Room> getRooms() {
         return rooms;
     }
+
+
+    @Override
+    public void execute(SensorEvent sensorEvent) {
+        RoomIteratorCollection roomIteratorCollection = new RoomIteratorCollectionImpl(rooms);
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%B8%D0%BC%D0%BF%D0%BB%D0%B5%D0%BC%D0%B5%D0%BD%D1%82%D0%B8%D1%80%D1%83%D0%B9%D1%82%D0%B5-%D0%B2%D0%BD%D1%83%D1%82%D1%80%D0%B5%D0%BD%D0%BD%D0%B8%D0%B9-%D0%B8%D1%82%D0%B5%D1%80%D0%B0%D1%82%D0%BE%D1%80